components/net/fetch/methods.rs

```diff
@@ -486,8 +488,20 @@ fn scheme_fetch(request: &mut Request,
                     Ok(file_path) => {
                         match File::open(file_path.clone()) {
                             Ok(mut file) => {
-                                let mut bytes = vec![];
-                                let _ = file.read_to_end(&mut bytes);
+                                let mut reader = BufReader::with_capacity(FILE_CHUNK_SIZE, file);
+                                let mut bytes = Vec::new();
+                                loop {
+                                    let length = {
+                                        let mut buffer = reader.fill_buf().unwrap().to_vec();
+                                        let buffer_len = buffer.len();
+                                        bytes.append(&mut buffer);
+                                        target.process_response_chunk(buffer);
```

Comment:
We actually have a system in place that requires less manual work, so let's make use of it. `wait_for_response` handles both synchronous and asynchronous responses; we can make this one asynchronous by:
1) spawning a new thread before we open the file
2) creating a new channel and storing it in `done_chan` (https://github.com/servo/servo/blob/f04c965a9bab46ff6943416baab338032bc526c1/components/net/http_loader.rs#L1091-L1093)
3) setting the body to ResponseBody::Receiving
4) sending the file chunks across the channel
5) setting the body to ResponseBody::Done once the file is completely read (https://github.com/servo/servo/blob/f04c965a9bab46ff6943416baab338032bc526c1/components/net/http_loader.rs#L1127-L1151)

To ensure everything still works, try loading a page from your local machine (not from a web server) that loads an image >32kb using a relative URL.

Reply:
Would `thread::spawn` work for making a new thread?

Reply:
Yes.

Reply:
Should I still call `target.process_response_chunk`?

Reply:
That should not be necessary in the model that I described.

Reply:
I had some time to work on this today, I'm building it to see if I can load files properly. I'll push the commit if it works.